src/redisearch_rs/trie_rs/src/iter/lending.rs

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+use super::{Iter, filter::TraversalFilter};
+use lending_iterator::prelude::*;
+use std::ffi::c_char;
+
+/// Iterates over the entries of a [`TrieMap`](crate::TrieMap) in lexicographical order, with minimal cloning.
+///
+/// Unlike [`Iter`], this iterator lets you borrow the current key, rather than having to clone it.
+///
+/// Invoke [`TrieMap::lending_iter`](crate::TrieMap::lending_iter) or
+/// [`TrieMap::prefixed_lending_iter`](crate::TrieMap::prefixed_lending_iter)
+/// to create an instance of this iterator.
+pub struct LendingIter<'tm, Data, F>(Iter<'tm, Data, F>);
+
+impl<'tm, Data, F> From<Iter<'tm, Data, F>> for LendingIter<'tm, Data, F> {
+    fn from(iter: Iter<'tm, Data, F>) -> Self {
+        LendingIter(iter)
```

Comment:
This will not avoid the [clone](https://github.com/RediSearch/RediSearch/blob/trie-iter-basic/src/redisearch_rs/trie_rs/src/iter/iter_.rs#L117) in the regular Iter, right?

Reply:
It does, actually!
Because `Iter::next` is never called by the lending iterator, we only invoke `Iter::advance`.